core/src/main/scala/cats/data/NonEmptySet.scala

```diff
@@ -346,6 +346,21 @@ sealed class NonEmptySetOps[A](val value: NonEmptySet[A]) {
     */
   def zipWithIndex: NonEmptySet[(A, Int)] =
     NonEmptySetImpl.create(toSortedSet.zipWithIndex)
+
+  /**
+    * Groups elements inside this `NonEmptySet` according to the `Order`
+    * of the keys produced by the given mapping function.
+    */
+  def groupBy[B](f: A => B)(implicit B: Order[B]): NonEmptyMap[B, NonEmptySet[A]] = {
```

Comment:
If we're returning `NonEmptyMap` here, does it make sense to make `NonEmptyList#groupBy` also return a non-empty map? @LukaJCB

Reply:
Yeah that's probably true, but we can't change it without breaking compatibility